File: frontend/lib/transaction/models/transaction.stats.dart
/// A class that holds stats about transactions
class TransactionStats {
  final double totalSpend;
  final double totalIncome;
  final double averageTransactionCost;
  final double largestExpense;
  final Map<String, int> categories;

  TransactionStats({
    required this.totalSpend,
    required this.totalIncome,
    required this.averageTransactionCost,
    required this.largestExpense,
    required this.categories,
  });

  factory TransactionStats.fromJson(Map<String, dynamic> json) {
    return TransactionStats(
      totalSpend: (json['totalSpend'] as num).toDouble(),
      totalIncome: (json['totalIncome'] as num).toDouble(),
      averageTransactionCost: (json['averageTransactionCost'] as num).toDouble(),
      largestExpense: (json['largestExpense'] as num).toDouble(),
      categories: Map<String, int>.from(json['categories']),
    );
  }
}

File: backend/src/model/api/transaction.stats.ts
import { Base } from "@backend/model/base";

/** A class for the transaction status request */
export class TransactionStatsRequest extends Base {
  /** How many days to look back for this request */
  days: number;

  constructor(days: number) {
    super();
    this.days = days;
  }
}

/** A class that holds stats about transactions */
export class TransactionStats extends Base {
  totalSpend: number;
  totalIncome: number;
  averageTransactionCost: number;
  largestExpense: number;
  categories: { [category: string]: number };

  constructor(totalSpend: number, totalIncome: number, averageTransactionCost: number, largestExpense: number, categories: TransactionStats["categories"]) {
    super();
    this.totalSpend = totalSpend;
    this.totalIncome = totalIncome;
    this.averageTransactionCost = averageTransactionCost;
    this.largestExpense = largestExpense;
    this.categories = categories;
  }
}
